### ISSUE TITLE
Contribute to k9 mail design

### ISSUE THREAD

**Issue by sedrubal** [scoute-dich/K9-MailClient#15]
It's a really good idea to modernize the k9mail design. But I think it would be better for all not to make a fork but to contribute to [k-9-design](https://github.com/k9mail/k-9-design). They're already planning and overworking the current design. 😉

**Comment by scoute-dich** [comment]
As I understand, they are working only at the ui concept. I think when when this is fine, they start coding. So at the moment I don't know how to contribute.

I just wanted a nice looking k9 without much work. So I simply changed the theme to material. I didn't thought, that this would become such a big thing...

**Comment by h-2** [comment]
Yeah, love this, but it would be much better if it were integrated upstream!
Why not make a pull-request and discuss things there?

**Comment by takiainen** [comment]
I'd like to see this happen aswell. Although in not a fan of current k9 design, all my emails are there, and as such a new app won't help of there isn't an easy way to transfer all the data.

**Comment by jomo** [comment]
@takiainen assuming your mails are on your mail servers as well, you can export and import settings with k9. You just need to provide the server passwords again.

**Comment by jancborchardt** [comment]
@scoute-dich so the changes you made, why not open pull requests for them in the original K9 Mail repository? :) https://github.com/k9mail/k-9

That would benefit everyone, and it doesn’t conflict with our work on the K9 design as a whole. :)

**Comment by scoute-dich** [comment]
Makes this sense, since I'm forced to use mind sdk21?

**Comment by takiainen** [comment]
since I'm not using Gmail, I have a very limited space on my email server and thus only store there as little as possible.

**Comment by jancborchardt** [comment]
@scoute-dich I’m not entirely sure about the technological aspects of it. :) The point is that K9 has a _lot_ of users, and lifting it up to material design would make it a lot better. And the K9 developers are very receptive and welcoming of better design.

So rather than having your own fork of it, why not have bigger impact? :) I’m actually working with one of the main developers to improve the general interaction design, as do other designers. Let’s collaborate, and if you happen to be in Berlin at some point, please let us know so we can work together. :)

**Comment by est42** [comment]
I have a suggestion as well, we forked K9 to include pretty easy privacy engine and material design in it. Check the code https://cacert.pep-security.lu/gitlab/android/pep 
It's free software and we welcome any contribution.

**Comment by jancborchardt** [comment]
@est42 same here: Why not open pull requests for them in the original K9 Mail repository? :) https://github.com/k9mail/k-9 – that would benefit everyone.

**Comment by est42** [comment]
@jancborchardt K9 dev team wasn't interesting about pEp. Afaik we already suggested pull requests but they were interested only about material design and nothing else.

pEp already have material design version for few months and now it's about catching up bugs and improving the design. It's not following the design style of K9, but instead making it easier for ordinary users. 

However since all projects are free software, anyone can take code from one and use it elsewhere. It benefits the users after all :)

**Comment by jancborchardt** [comment]
It doesn’t really benefit users if it doesn’t reach them. :) Current K9 has millions of users, but does not have Material Design.

> Afaik we already suggested pull requests but they were interested only about material design and nothing else.

That’s ok though, right? Did you split up the pull requests so the ones with Material Design could be reviewed and merged separately? Splitting it up and making it easy to merge small improvements is key.

**Comment by est42** [comment]
@jancborchardt we don't have enough resources to do that. We are trying to make solution, which would be different than K9 and offer not only end2end encryption( pluggable encryption like PGP, signal and others) but also anonymous transport of emails over GNUnet. That's not compatible with K9. The advantage is that pEp will be available for multiple platforms and not limited to Android. 

However, anyone can just take Material Design pieces from pEp for Android and just use it. Code is free and public. So if you have a time, just do it :)
